internal/printer/printer.go

```diff
@@ -1404,7 +1404,11 @@ func (p *Printer) emitTypeParameter(node *ast.TypeParameterDeclaration) {
 }
 
 func (p *Printer) emitTypeParameterNode(node *ast.TypeParameterDeclarationNode) {
-	p.emitTypeParameter(node.AsTypeParameter())
+	if ast.IsTypeParameterDeclaration(node) {
+		p.emitTypeParameter(node.AsTypeParameter())
+	} else {
+		p.emitTypeArgument(node)
+	}
```

Comment:
This felt spooky until I read the original code:

```
function emitTypeParameters(parentNode: SignatureDeclaration | InterfaceDeclaration | TypeAliasDeclaration | ClassDeclaration | ClassExpression, typeParameters: NodeArray<TypeParameterDeclaration> | undefined) {
    if (isFunctionLike(parentNode) && parentNode.typeArguments) { // Quick info uses type arguments in place of type parameters on instantiated signatures
        return emitTypeArguments(parentNode, parentNode.typeArguments);
    }
    emitList(parentNode, typeParameters, ListFormat.TypeParameters | (isArrowFunction(parentNode) ? ListFormat.AllowTrailingComma : ListFormat.None));
}
```

Can we leave a comment here saying why it's okay that this function is attempting to handle the wrong kind of thing?

Reply:
Yeah, it's a bit ugly, but way less ugly than what we used to do. The main issue is that we're wanting to display a declaration, but with type arguments instead of type parameters--and type arguments only occur in calls. Allowing type arguments instead of type parameters in node building and emit seems like a reasonable way to accomplish this.

Reply:
Anyway, I'll add a comment.